### PR TITLE
Add document limits to index and bulk pages

### LINKED ISSUE #4391: [DOC] Update the document ID limit in the documentation
**What do you want to do?**
 
- [X] Request a change to existing documentation
- [ ] Add new documentation
- [x] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.

 Issue 1: There is a limit of 512 bytes for Document ID but this isn't mentioned anywhere in the documentation anywhere. 
Issue 2: This limit is not being honoured in the bulk updates ([tracking bug](https://github.com/opensearch-project/OpenSearch/issues/6595))

We need to:
- Update the documentation with the `_id` limits ([index data](https://opensearch.org/docs/1.2/opensearch/index-data/) and [bulk](https://opensearch.org/docs/1.2/opensearch/rest-api/document-apis/bulk/)). 
- Call out that (2.9 onwards), we're enforcing this limit during bulk updates as well.

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Fixes #4391 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _api-reference/document-apis/bulk.md [hunk 1]
```diff
@@ -14,6 +14,9 @@ Introduced 1.0
 The bulk operation lets you add, update, or delete multiple documents in a single request. Compared to individual OpenSearch indexing requests, the bulk operation has significant performance benefits. Whenever practical, we recommend batching indexing operations into bulk requests.
 
 
+Beginning in OpenSearch 2.9, the bulk operation will contain a memory limit for each document in the request of 512mb or less.
```
Comment (Naarcha-AWS) [inline _api-reference/document-apis/bulk.md:17]:
```suggestion
Beginning in OpenSearch 2.9, the bulk operation will contain a memory limit for each document in the request of 512 mb or less.
```

File: _im-plugin/index.md [hunk 2]
```diff
@@ -16,6 +16,8 @@ You index data using the OpenSearch REST API. Two APIs exist: the index API and
 
 For situations in which new data arrives incrementally (for example, customer orders from a small business), you might use the index API to add documents individually as they arrive. For situations in which the flow of data is less frequent (for example, weekly updates to a marketing website), you might prefer to generate a file and send it to the `_bulk` API. For large numbers of documents, lumping requests together and using the `_bulk` API offers superior performance. If your documents are enormous, however, you might need to index them individually.
 
+To make sure that many documents can be indexed through both index API and bulk API are manageable, each document inside an index must be less than 512mb.
```
Comment (Naarcha-AWS) [inline _im-plugin/index.md:19]:
```suggestion
To make sure that many documents can be indexed through both the index API and the bulk API are manageable, each document inside an index must be less than 512 mb. 
```

File: _api-reference/document-apis/bulk.md [hunk 3]
```diff
@@ -14,6 +14,9 @@ Introduced 1.0
 The bulk operation lets you add, update, or delete multiple documents in a single request. Compared to individual OpenSearch indexing requests, the bulk operation has significant performance benefits. Whenever practical, we recommend batching indexing operations into bulk requests.
 
 
+Beginning in OpenSearch 2.9, the bulk operation will contain a memory limit for each document in the request of 512 mb or less.
```
Comment (Naarcha-AWS) [inline _api-reference/document-apis/bulk.md:17]:
```suggestion
Beginning in OpenSearch 2.9, the bulk operation will contain a memory limit for the document `_id` in the request of 512 MB or less.
```

File: _im-plugin/index.md [hunk 4]
```diff
@@ -16,6 +16,8 @@ You index data using the OpenSearch REST API. Two APIs exist: the index API and
 
 For situations in which new data arrives incrementally (for example, customer orders from a small business), you might use the index API to add documents individually as they arrive. For situations in which the flow of data is less frequent (for example, weekly updates to a marketing website), you might prefer to generate a file and send it to the `_bulk` API. For large numbers of documents, lumping requests together and using the `_bulk` API offers superior performance. If your documents are enormous, however, you might need to index them individually.
 
+When indexing documents, the document `_id` must be less than 512 MB.
```
Comment (natebower) [inline _im-plugin/index.md:19]:
```suggestion
When indexing documents, the document `_id` must be 512 MB or less in size.
```

File: _api-reference/document-apis/bulk.md [hunk 5]
```diff
@@ -14,6 +14,9 @@ Introduced 1.0
 The bulk operation lets you add, update, or delete multiple documents in a single request. Compared to individual OpenSearch indexing requests, the bulk operation has significant performance benefits. Whenever practical, we recommend batching indexing operations into bulk requests.
 
 
+Beginning in OpenSearch 2.9, the bulk operation will contain a memory limit for the document `_id` in the request of 512 MB or less.
```
Comment (natebower) [inline _api-reference/document-apis/bulk.md:17]:
```suggestion
Beginning in OpenSearch 2.9, when indexing documents using the bulk operation, the document `_id` must be 512 MB or less in size.
```

**Review by natebower — APPROVED** [review]
LGTM with noted changes. Thanks!

**Pull request merged**